persist/schema/types.go

```diff
@@ -20,14 +20,33 @@
 
 package schema
 
+// MajorVersion is the major schema version for a set of fileset files,
+// this is only incremented when breaking changes are introduced and
+// tooling needs to upgrade older files to newer files before a server restart
+const MajorVersion = 1
+
 // IndexInfo stores metadata information about block filesets
 type IndexInfo struct {
-	Start     int64
-	BlockSize int64
-	Entries   int64
+	Start        int64
+	BlockSize    int64
+	Entries      int64
+	MajorVersion int64
+	Summaries    IndexSummariesInfo
+	BloomFilter  IndexBloomFilterInfo
+}
+
+// IndexSummariesInfo stores metadata about the summaries
+type IndexSummariesInfo struct {
+	Summaries int64
```

Comment:
Should this be `NumSummaries`?

Reply:
To match the `Entries int64` it does not have the num prefix.

Reply:
sounds good